akka-projection-jdbc/src/test/scala/akka/projection/jdbc/JdbcProjectionSpec.scala

```diff
@@ -274,16 +275,28 @@ class JdbcProjectionSpec
     }
   }
 
+  private def assertOffsetIs(
+      expected: Long)(implicit offsetStore: JdbcOffsetStore[PureJdbcSession], projectionId: ProjectionId) = {
+    val offset = offsetStore.readOffset[Long](projectionId).futureValue.value
+    offset shouldBe expected
+  }
```

Comment:
I didn't bring the `withClue` in this method to see how its usages would look like.

Reply:
I think `withClue` here is more interesting than on the `assertOffsetIsEmpty` one. Because each time we have different expected value.

On the other hand, the right way of using it is to have different clues on each call. Like:

```scala
assertOffsetIs(4L, "after failing the source")
...
assertOffsetIs(6L, "after processing the whole stream")
```

Reply:
My opinion about `withClue` is that it's not adding any more information than what is already in the stack trace. I didn't comment on that when we started the project because I thought it wasn't important, but if we are making major refactoring of tests we should consider removing the boilerplate clues altogether. That said, if you find it important to include I don't mind.